src/Groups/GAPGroups.jl

```diff
@@ -0,0 +1,617 @@
+The parent of `x` is set as Sym(n), where n is the largest moved point of `x`.
+"""
+function cperm(L::Union{Array{Int64,1},UnitRange{Int64}}...)
+   if length(L)==0
+      return one(symmetric_group(1))
+   else
+      return prod([PermGroupElem(symmetric_group(maximum(y)), GAP.Globals.CycleFromList(GAP.julia_to_gap(collect(y)))) for y in L])
+   end
+end
+
+# cperm stays for "cycle permutation", but we can change name if we want
+# takes as input a list of arrays (not necessarly disjoint)
+# WARNING: we allow e.g. PermList([2,3,1,4,5,6]) in Sym(3)
+
+"""
+    cperm(G::PermGroup, L::Union{Array{Int64,1},UnitRange{Int64}}...)
+For given lists of positive integers `[a_1, a_2, ..., a_n],[b_1, b_2, ... , b_m], ...` return the
+permutation `(a_1,a_2,...,a_n)(b_1,b_2,...,b_m)...`. The array `[n,n+1,...,n+k]` can be replaced by `n:n+k`.
+  
+If a list is epmty or contains duplicates or holes, it fails.
+The parent of `x` is G. If `x` is not in G, it return ERROR.
+"""
+function cperm(g::PermGroup,L::Union{Array{Int64,1},UnitRange{Int64}}...)
+   if length(L)==0
+      return one(g)
+   else
+      x=GAP.Globals.Product(GAP.julia_to_gap([GAP.Globals.CycleFromList(GAP.julia_to_gap(collect(y))) for y in L]))
+      if GAP.Globals.IN(x,g.X) return PermGroupElem(g, x)
+      else throw(ArgumentError("the element does not embed in the group"))
+      end
+   end
+end
+
+"""
+    listperm(x::PermGroupElem)
+is the list L defined by L = [ `x`(i) for i in 1:n ], where Sym(n) = parent(`x`).
+"""
+function listperm(x::PermGroupElem)
+   return [x(i) for i in 1:x.parent.deg]
+end
+
+"""
+    gens(G::Group)
+return an array of generators of the group `G`.
+"""
+function gens(G::Group)
+   L = GAP.Globals.GeneratorsOfGroup(G.X)
+   res = Vector{elem_type(G)}(undef, length(L))
+   for i = 1:length(res)
+     res[i] = group_element(G, L[i])
+   end
+   return res
+end
+
+"""
+
+    gens(G::Group, i::Int)
+return the `i`-th element of the array gens(`G`). If `i` is greater than the length of gens(`G`), an ERROR is returned.
+"""
+function gens(G::Group, i::Int)
+   L = GAP.Globals.GeneratorsOfGroup(G.X)
+   @assert length(L) >= i "The number of generators is lower than the given index"
+   return group_element(G, L[i])
+end
+
+"""
+    ngens(G::Group) -> Int
+return the length of the array gens(G).
+
+!!! warning "WARNING:" 
+    this is *NOT*, in general, the minimum number of generators for G.
+"""
+ngens(G::Group) = length(GAP.Globals.GeneratorsOfGroup(G.X))
+
+
+Base.getindex(G::Group, i::Int) = gens(G, i)
+Base.sign(x::PermGroupElem) = GAP.Globals.SignPerm(x.X)
+
+Base.isless(x::PermGroupElem, y::PermGroupElem) = x<y
+
+#embedding of a permutation in permutation group
+function (G::PermGroup)(x::PermGroupElem)
+   if !GAP.Globals.IN(x.X,G.X)
+      throw(ArgumentError("the element does not embed in the group"))
+   end
+   return group_element(G, x.X)
+end
+
+#evaluation function
+function (x::PermGroupElem)(n)
```

Comment:
`Integer`?

Reply:
You mean this?
```suggestion
function (x::PermGroupElem)(n::Integer)
```
Seems sensible to me.

Reply:
yes; also note that there is already a precedence:
https://github.com/Nemocas/AbstractAlgebra.jl/blob/master/src/generic/PermGroups.jl#L42

which, when properly extended makes working with permutations of different degrees very friendly

Reply:
Right.